mmpy_bot/plugins/base.py

```diff
@@ -88,21 +70,213 @@ async def call_function(
             # a plugin-specific thread or process pool if we wanted.
             self.driver.threadpool.add_task(function, event, *groups)
 
-    def get_help_string(self):
-        string = f"Plugin {self.__class__.__name__} has the following functions:\n"
-        string += "----\n"
-        for functions in self.message_listeners.values():
-            for function in functions:
-                string += f"- {function.get_help_string()}"
-            string += "----\n"
-        if len(self.webhook_listeners) > 0:
-            string += "### Registered webhooks:\n"
-            for functions in self.webhook_listeners.values():
-                for function in functions:
-                    string += f"- {function.get_help_string()}"
+
+class HelpPlugin(Plugin):
+    """Provide a `help` command that lists functions provided by all plugins.
+
+    With a few plugins enabled the help text can become quite verbose. For this reason,
+    this plugin defaults to sending a private/direct message with the information.
+
+    If you wish to disable this behavior pass `direct_help=False` and the help text will
+    be displayed in the channel where it is requested.
+    """
+
+    def __init__(
+        self,
+        direct_help: bool = True,
+    ):
+        super().__init__()
+        self.direct_help: bool = direct_help
+
+    def initialize(
+        self,
+        driver: Driver,
+        plugin_manager: PluginManager,
+        settings: Settings,
+    ):
+        super().initialize(driver, plugin_manager, settings)
+
+        if self.settings.RESPOND_CHANNEL_HELP:
+            self.help = listen_to("^!help$")(self.help)
+
+    def get_help_string(self, message: Message) -> str:
```

Comment:
Given the very similar name to `get_help` below, can you add a docstring here?

Reply:
Done